From 7dec562d1d5723b7eea08d08659dc9dc4e79c8ca Mon Sep 17 00:00:00 2001
From: David Abdurachmanov <davidlt@cern.ch>
Date: Wed, 11 Mar 2015 11:35:47 +0100
Subject: [PATCH 1/1] Update CINT CMake and configure scripts for GCC 5

The following is required otherwise `iosenum.h` will be missing causing
build failures. Needed for Fedora 22.

Signed-off-by: David Abdurachmanov <davidlt@cern.ch>
---
 cint/cint/CMakeLists.txt |  4 +---
 cint/cint/Module.mk      | 12 ++++++++++++
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/cint/cint/CMakeLists.txt b/cint/cint/CMakeLists.txt
index 705b57d..e7756be 100644
--- a/cint/cint/CMakeLists.txt
+++ b/cint/cint/CMakeLists.txt
@@ -61,9 +61,7 @@ set_source_files_properties(loadfile_tmp.cxx  COMPILE_FLAGS "-UR__HAVE_CONFIG -D
 #---Generate and/or copy the iosenum headerfile to the include directory-------------------------------
 if(${CLANG_MAJOR} GREATER 2)
   set(IOSENUM_H ${PROJECT_SOURCE_DIR}/cint/iosenum/iosenum.${ROOT_ARCHITECTURE}3)
-elseif(${GCC_MAJOR} EQUAL 4)
-  set(IOSENUM_H ${PROJECT_SOURCE_DIR}/cint/iosenum/iosenum.${ROOT_ARCHITECTURE}3)
-elseif(${GCC_MAJOR} EQUAL 3)
+elseif(${GCC_MAJOR} GREATER 3)
   set(IOSENUM_H ${PROJECT_SOURCE_DIR}/cint/iosenum/iosenum.${ROOT_ARCHITECTURE}3)
 else()
   set(IOSENUM_H ${PROJECT_SOURCE_DIR}/cint/iosenum/iosenum.${ROOT_ARCHITECTURE})
diff --git a/cint/cint/Module.mk b/cint/cint/Module.mk
index 4cfed92..0293321 100644
--- a/cint/cint/Module.mk
+++ b/cint/cint/Module.mk
@@ -161,6 +161,10 @@ ifeq ($(GCC_MAJOR),4)
 CINTS2       := $(filter-out $(MODDIRSD)/libstrm.%,$(CINTS2))
 CINTS2       += $(MODDIRSD)/gcc4strm.cxx
 endif
+ifeq ($(GCC_MAJOR),5)
+CINTS2       := $(filter-out $(MODDIRSD)/libstrm.%,$(CINTS2))
+CINTS2       += $(MODDIRSD)/gcc4strm.cxx
+endif
 ifneq ($(CLANG_MAJOR),)
 CINTS2       := $(filter-out $(MODDIRSD)/libstrm.%,$(CINTS2))
 CINTS2       += $(MODDIRSD)/gcc4strm.cxx
@@ -203,6 +207,9 @@ IOSENUMC     := $(CINTDIRIOSEN)/iosenum.cxx
 ifneq ($(CLANG_MAJOR),)
 IOSENUMA     := $(CINTDIRIOSEN)/iosenum.$(ARCH)3
 else
+ifeq ($(GCC_MAJOR),5)
+IOSENUMA     := $(CINTDIRIOSEN)/iosenum.$(ARCH)3
+else
 ifeq ($(GCC_MAJOR),4)
 IOSENUMA     := $(CINTDIRIOSEN)/iosenum.$(ARCH)3
 else
@@ -213,6 +220,7 @@ IOSENUMA     := $(CINTDIRIOSEN)/iosenum.$(ARCH)
 endif
 endif
 endif
+endif
 
 # used in the main Makefile
 ALLHDRS     += $(CINTHT) $(CINTINCLUDES)
@@ -344,6 +352,10 @@ $(call stripsrc,$(CINTDIRSD)/%strm.o):    CINTCXXFLAGS += -I$(CINTDIRL)/$(notdir
 ifeq ($(GCC_MAJOR),4)
 $(call stripsrc,$(CINTDIRSD)/gcc4strm.o): CINTCXXFLAGS += -Wno-strict-aliasing
 endif
+ifeq ($(GCC_MAJOR),5)
+$(call stripsrc,$(CINTDIRSD)/gcc4strm.o): CINTCXXFLAGS += -Wno-strict-aliasing
+endif
+
 
 $(MAKECINTO) $(CINTO): $(CINTCONF) $(ORDER_) $(CINTINCLUDES)
 
-- 
1.8.2

